File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/StreamingReadCombine.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.Fragment.*

@SuppressWarnings(Array("unused"))
object StreamingReadCombine:
  var tx: Transactor = null // placeholder

  //start
  // Open two cursors simultaneously on the same connection
  def mergedNames(): List[String] =
    val activeUsers = Fragment.of("SELECT name FROM users WHERE active")
      .streamingQuery(PgTypes.text, 512)
    val archivedUsers = Fragment.of("SELECT name FROM archived_users")
      .streamingQuery(PgTypes.text, 512)

    activeUsers.combine(archivedUsers).map { (active, archived) =>
      active.toList ++ archived.toList
    }.transact(tx)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/StreamingReadCombine.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.*;
import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("unused")
public class StreamingReadCombine {
  Transactor tx = null; // placeholder

  // start
  // Open two cursors simultaneously on the same connection
  List<String> mergedNames() {
    var activeUsers =
        Fragment.of("SELECT name FROM users WHERE active").streamingQuery(PgTypes.text, 512);
    var archivedUsers =
        Fragment.of("SELECT name FROM archived_users").streamingQuery(PgTypes.text, 512);

    return activeUsers
        .combine(archivedUsers)
        .map(
            cursors -> {
              List<String> all = new ArrayList<>();
              all.addAll(cursors._1().toList());
              all.addAll(cursors._2().toList());
              return all;
            })
        .transact(tx);
  }
  // stop
}
